coding/cpp_style_guide.rst

```diff
@@ -1242,6 +1242,36 @@ A class or struct definition must explicitly declare the privacy qualifier of it
 Although C++ provides the above definitions as defaults, some compilers generate warnings if explicit privacy qualifiers are not specified.
 This Rule will reduce unnecessary compiler warnings.
 
+.. _style-guide-cpp-5-2c:
+
+5-2c. Classes SHOULD explicitly declare or delete compiler-generated members
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All new C++ classes SHOULD explicitly default or delete all constructors and assignment operators that may be synthesized by the compiler.
+Similarly, destructors should be explicitly defined using the ``default`` keyword as well when no custom implementation is needed.
+In practice, that means all classes should explicitly define the following five members using either ``default``, ``delete``, or a custom implementation (shown here defaulted):
+
+.. code-block:: cpp
+
+    class Foo {
+    public:
+        ...
+        Foo(Foo const &) = default;
+        Foo(Foo &&) = default;
+        Foo & operator=(Foo const &) = default;
+        Foo & operator=(Foo &&) = default;
+        ~Foo() = default;
+        ...
+    };
+
+Defining these members forces us to consider their behavior and judge whether the default behavior is in fact desired when implementing a new class.
+It also makes compilation errors appear sooner (rather than delaying them until a synthesized member is first needed).
+
+When these members cannot be defaulted or deleted, a comment explaining why SHOULD accompany the implementation.
```

Comment:
Should a note be added for `delete` also?  That would also help to interrupt the reader who might otherwise not notice the distinction between medium-length words that start with "de" and will be more boilerplate-ish given this guideline.  In many cases it seems to me that custom implementations are straightforward and don't necessarily need explanation, but I'm OK with asking for one anyway.

Reply:
Done.  I believe @r-owen liked the idea of asking for comments when using `delete`, too, and while I do feel like there are times when it's perfectly clear without a comment, I certainly agree that there are times when it isn't.